semgrep/semgrep/error.py

```diff
@@ -342,6 +343,47 @@ def to_dict_base(self) -> Dict[str, Any]:
         }
 
 
+@attr.s(frozen=True, eq=True)
+class CoreWarning(SemgrepError):
+    check_id: str = attr.ib()
+    msg: str = attr.ib()
+
+    code = INVALID_PATTERN_EXIT_CODE
```

Comment:
@brendongo Despite this Semgrep still returns 0 when one of these warnings is reported, is that normal?

Reply:
I'll merge but let me know if I should do something about this.